backend/app/Models/User.php

```diff
@@ -175,15 +175,33 @@ public function publications(): BelongsToMany
     }
 
     /**
-     * @return \App\Models\Role
+     * Return the highest privileged role ID for the user in the following order:
+     * 1. Application Administrator
+     * 2. Publication Administrator
+     * 3. Editor
+     * 4. Review Coordinator
+     * 5. Reviewer
+     * 6. Submitter
+     *
+     * @return int|null
      */
-    public function getHighestPrivilegedRole()
+    public function getHighestPrivilegedRole(): int|null
     {
-        return $this->roles->sortBy('id')->first();
+        if ($this->hasRole(Role::APPLICATION_ADMINISTRATOR)) {
+            return (int)Role::APPLICATION_ADMINISTRATOR_ROLE_ID;
+        }
+        if ($this->publications->isNotEmpty()) {
+            return PublicationUser::where('user_id', $this->id)->get(['role_id'])->sort()->first()->role_id;
```

Comment:
```suggestion
            return PublicationUser::where('user_id', $this->id)->min('role_id');
```

This feels a little more clear, but more importantly doesn't fetch all the rows to find the minimum.